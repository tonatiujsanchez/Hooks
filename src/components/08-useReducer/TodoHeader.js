import useForm from "../../hooks/useForm"


const TodoHeader = ({ todos, handleAddTodo }) => {

    const [{ description }, handleInputChange, resetForm] = useForm({
        description: ''
    })

    const handleSubmit = ( e ) =>{
        e.preventDefault()

        if (description.trim() === '') {
            return
        }

        const newTodo = {
            id: Date.now(),
            desc: description,
            done: false
        }

        handleAddTodo( newTodo )

        resetForm()

    }

    return (
        <header className='d-flex justify-content-between align-items-center flex-wrap'>
            <h1 className='fw-bold fs-2 m-0'>TODO APP <small className='fs-5'>( {todos.length} )</small></h1>
            <form onSubmit={handleSubmit} className='d-flex gap-2 align-items-center'>
                <input
                    type="text"
                    name='description'
                    className='form-control'
                    placeholder='Nueva Tarea'
                    autoComplete='off'
                    value={description}
                    onChange={handleInputChange} />
                <button type='submit' className='btn btn-outline-success'>Agregar</button>
            </form>
        </header>
    )
}

export default TodoHeader