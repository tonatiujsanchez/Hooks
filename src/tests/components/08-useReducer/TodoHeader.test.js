import { shallow } from "enzyme"
import TodoHeader from "../../../components/08-useReducer/TodoHeader"
import { demoTodos } from "../../fixtures/demoTodos"

describe('Pruebas en <TodoHeader/>', () => { 

    const handleAddTodo = jest.fn()

    const wrapper = shallow( 
        <TodoHeader 
            todos={ demoTodos } 
            handleAddTodo={ handleAddTodo } 
        />
    )

    test('Debe de mostrarse correctamente', () => {

        expect( <handleAddTodo /> ).toMatchSnapshot()

    })

    test('No debe de llamar AddTodo', () => { 

        const formSubmit = wrapper.find('form').prop('onSubmit')
        formSubmit( { preventDefault(){} } )

        expect( handleAddTodo ).not.toBeCalled()

     })

     test('Debe de llamar la funcion handleAddTodo', () => { 

        const inputChange = wrapper.find('input').prop('onChange')
        inputChange({target:{ name:'description', value:'Hola mundo!' }})


        const formSubmit = wrapper.find('form').prop('onSubmit')
        formSubmit( { preventDefault(){} } )


        expect( handleAddTodo ).toHaveBeenCalledTimes(1)
        expect( handleAddTodo ).toHaveBeenCalledWith(expect.any(Object))


        const todoTest = {
            id: expect.any(Number),
            desc: 'Hola mundo!',
            done: false
        }

        expect( handleAddTodo ).toHaveBeenCalledWith(todoTest)


        const inputValue = wrapper.find('input').prop('value')
        expect( inputValue ).toBe('')
        
      })

 })