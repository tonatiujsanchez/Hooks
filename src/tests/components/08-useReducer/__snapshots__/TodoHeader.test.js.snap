// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoHeader/> Debe de mostrarse correctamente 1`] = `<handleAddTodo />`;
